gossip/state/state.go

```diff
@@ -780,6 +809,11 @@ func (s *GossipStateProviderImpl) hasRequiredHeight(height uint64) func(peer dis
 
 // AddPayload adds new payload into state.
 func (s *GossipStateProviderImpl) AddPayload(payload *proto.Payload) error {
+
+	if !s.CanProcessPayloadFromOrderer() {
```

Comment:
Instead of CanProcessPayloadFromOrderer, call a an extension, e.g. s.ext.AddPayload.

Reply:
sounds good, will do that

Reply:
done